### PR TITLE
Feature/transform to tf

### PR DESCRIPTION
**Pull request by alexmillane**
Moved from iros drone race (with history) to a more sensible location.

### CODE REVIEW

**Comment by alexmillane** [comment]
@ZacharyTaylor @helenol

File: transform_to_tf/include/transform_to_tf/transform_to_tf.hpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  Transformation transformation;
+};
+
+// Class handling global alignment calculation and publishing
+class TransformToTf {
+ public:
+  // Constructor
+  TransformToTf(const ros::NodeHandle& nh, const ros::NodeHandle& nh_private);
+
+ protected:
```
Comment (ZacharyTaylor) [inline transform_to_tf/include/transform_to_tf/transform_to_tf.hpp]: Why protected, are you intending to need all of this exposed in inheriting classes?

File: transform_to_tf/include/transform_to_tf/transform_to_tf.hpp [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  ros::NodeHandle nh_private_;
+
+  // Subscribers
+  ros::Subscriber transform_sub_;
+
+  // TF publishing timer and broadcaster object.
+  tf::TransformBroadcaster tf_broadcaster_;
+  ros::Timer tf_timer_;
+
+  // Stores the position of the current hoop in the world frame
```
Comment (ZacharyTaylor) [inline transform_to_tf/include/transform_to_tf/transform_to_tf.hpp]: update comment now function is more generic

File: transform_to_tf/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,24 @@
+cmake_minimum_required(VERSION 2.8.0)
+project(transform_to_tf)
+
+add_definitions(-std=c++11)
+
+find_package(catkin_simple REQUIRED)
+catkin_simple(ALL_DEPS_REQUIRED)
+
+find_package(Eigen REQUIRED)
```
Comment (ZacharyTaylor) [inline transform_to_tf/CMakeLists.txt:9]: Why the preference for the eigen system library?

File: transform_to_tf/src/library/transform_to_tf.cpp [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+      global_frame_name_(kDefaultGlobalFrameName) {
+  // Getting data and params
+  subscribeToTopics();
+  advertiseTopics();
+  getParametersFromRos();
+}
+
+void TransformToTf::subscribeToTopics() {
+  // Subscribing to the required data topics
+  transform_sub_ = nh_.subscribe("transform_to_be_converted", 1,
```
Comment (ZacharyTaylor) [inline transform_to_tf/src/library/transform_to_tf.cpp:20]: Might make sense to have a queue size larger then 1

File: transform_to_tf/src/library/transform_to_tf.cpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                                 &TransformToTf::transformCallback, this);
+}
+
+void TransformToTf::advertiseTopics() {
+  // Advertising topics
```
Comment (ZacharyTaylor) [inline transform_to_tf/src/library/transform_to_tf.cpp:25]: Not sure advertise is the best description of what is happening in this function

File: transform_to_tf/src/library/transform_to_tf.cpp [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  tf_timer_ = nh_.createTimer(ros::Duration(0.01),
+                              &TransformToTf::publishTFTransform, this);
+}
+
+void TransformToTf::getParametersFromRos() {
+  // Retrieving the parameters
+  nh_private_.getParam("verbose", verbose_);
```
Comment (ZacharyTaylor) [inline transform_to_tf/src/library/transform_to_tf.cpp:32]: I prefer the use of .param over getParam, as while it works I don't think that getParam is specified to not modify the variable when it hasn't found a value in the parameter server.

File: transform_to_tf/src/transform_to_tf_node.cpp [hunk 7]
```diff
@@ -0,0 +1,20 @@
+#include <ros/ros.h>
+#include <glog/logging.h>
+#include "transform_to_tf/transform_to_tf.hpp"
+
+// Standard C++ entry point
+int main(int argc, char** argv) {
+  // Starting the logging
+  google::InitGoogleLogging(argv[0]);
```
Comment (ZacharyTaylor) [inline transform_to_tf/src/transform_to_tf_node.cpp]: Unless I missed it you never actually use the logging

**Review by ZacharyTaylor — APPROVED** [review]
scolgtm

File: transform_to_tf/CMakeLists.txt (hunk 3 above)
Comment (alexmillane) [reply transform_to_tf/CMakeLists.txt:9]: I have no good answer for that other than to be consistent with most of our other packages which seem to use the system lib... My preference is to depend only catkinized packages.

File: transform_to_tf/include/transform_to_tf/transform_to_tf.hpp (hunk 1 above)
Comment (alexmillane) [reply transform_to_tf/include/transform_to_tf/transform_to_tf.hpp]: In general I have been making things protected to give as much flexibility to inheriting classes. But yeah, I checked the style guide and says to make things private except where needed. Good call.

File: transform_to_tf/src/transform_to_tf_node.cpp (hunk 7 above)
Comment (alexmillane) [reply transform_to_tf/src/transform_to_tf_node.cpp]: Good spotting. thx

**Pull request merged**